### PR TITLE
chore: refactor job as global

### PR DESCRIPTION
**Pull request by erikwilson**
## Description

Part 1 of job refactor. Redefine job actor as global and use system for actor context.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by maxrussell — COMMENTED** [review]
Just some super minor nits. Should be ready for @stoksc's eyes.

File: master/internal/core.go [hunk 1]
```diff
@@ -1010,11 +1009,12 @@ func (m *Master) Run(ctx context.Context) error {
 		},
 		cert,
 	)
+	job.Manager = job.NewJobs(m.rm, m.system)
```
Comment (stoksc) [inline master/internal/core.go:1012]: NewManager?

File: master/internal/job/job.go [hunk 2]
```diff
@@ -4,30 +4,39 @@ import (
...
 	"github.com/determined-ai/determined/master/internal/rm"
 	"github.com/determined-ai/determined/master/internal/sproto"
 	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/master/pkg/model"
-	"github.com/determined-ai/determined/proto/pkg/apiv1"
 	"github.com/determined-ai/determined/proto/pkg/jobv1"
 )
 
+// Manager is the global singleton for managing jobs.
+var Manager *Jobs
```
Comment (stoksc) [inline master/internal/job/job.go:22]: i might rename this to `DefaultManager` and rename `Jobs` to `Manager` but is only an opinion

File: master/internal/job/job.go [hunk 3]
```diff
@@ -124,134 +137,117 @@ func (j *Jobs) getJobs(
...
 }
 
-func (j *Jobs) setJobPriority(ctx *actor.Context, jobID model.JobID, priority int) error {
+func (j *Jobs) setJobPriority(ref *actor.Ref, priority int) error {
 	if priority < 1 || priority > 99 {
 		return errors.New("priority must be between 1 and 99")
 	}
-	jobActor := j.actorByID[jobID]
-	resp := ctx.Ask(jobActor, sproto.SetGroupPriority{
+	resp := j.system.Ask(ref, sproto.SetGroupPriority{
```
Comment (stoksc) [inline master/internal/job/job.go:144]: the only thing weird about this is that if the actor is closed, it just returns nil, but that is probably fine behavior for this anyway

File: master/internal/job/job.go [hunk 4]
```diff
@@ -124,134 +137,117 @@ func (j *Jobs) getJobs(
...
-		} else if len(errors) > 1 {
-			ctx.Respond(fmt.Errorf("encountered the following errors: %s", strings.Join(errors, ", ")))
-		}
-	default:
-		return actor.ErrUnexpectedMessage(ctx)
+	}
+	if len(errors) == 1 {
+		return fmt.Errorf(errors[0])
+	} else if len(errors) > 1 {
+		return fmt.Errorf("encountered the following errors: %s", strings.Join(errors, ", "))
```
Comment (stoksc) [inline master/internal/job/job.go:250]: hashicorp multierror? :shrug:

**Review by stoksc — APPROVED** [review]
minor comments also, this looks great

File: master/internal/job/job.go (hunk 4 above)
Comment (erikwilson) [reply master/internal/job/job.go:250]: yeah, I think the error handling could be improved here. Continuing on job not found and using `errors.Join` (or a type that implements `Unwrap() []error`) might be preferable. My main concern was preserving current behavior but can update as another PR.

File: master/internal/api_experiment.go [hunk 5]
```diff
@@ -264,12 +264,7 @@ func (a *apiServer) GetExperiment(
 	}
 
 	jobID := model.JobID(exp.JobId)
-
-	jobSummary := &jobv1.JobSummary{}
-	err = a.ask(sproto.JobsActorAddr, sproto.GetJobSummary{
```
Comment (stoksc) [inline master/internal/api_experiment.go:269]: can we remove the `sproto.GetJobSummary` type yet?

**Review by stoksc — APPROVED** [review]
lgtm modulo tests passing

**Pull request merged**